src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -103,7 +104,9 @@ def build_expr_from_python_val(expr_builder: proto.Expr, obj: Any) -> None:
         unscaled_val = reduce(lambda val, digit: val * 10 + digit, dec_tuple.digits)
         if dec_tuple.sign != 0:
             unscaled_val *= -1
-        req_bytes = (unscaled_val.bit_length() + 7) // 8
+        # Need to round up.
+        req_bytes = int(math.ceil((unscaled_val.bit_length() + 7) / 8))
```

Comment:
I'm not sure I understand. If using FP division and `math.ceil`, what's the point of `+ 7`?

Reply:
It appears the argument to `ceil` is in the integer domain. `... / 8.0` should fix that and eliminate the need for `+7`.

Reply:
refactored and added comment.